quinn/src/notify.rs

```diff
@@ -0,0 +1,89 @@
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Waker};
+
+use slab::Slab;
+
+/// Broadcasts an event to any number of waiters
+#[derive(Clone)]
+pub struct NotifyOwned {
```

Comment:
You recall correctly. Tokio's `Notify` is appealing, but because `Notified` borrows from the `Notify` it cannot be easily used to implement `'static` futures, which I hypothesized might be useful downstream, based on https://github.com/hyperium/h3/pull/80 and https://github.com/libp2p/rust-libp2p/pull/2289#issuecomment-1119872606. However, I haven't been able to reach any of the h3 folks to discuss this in recent months, so some question remains as to whether it's really useful. We should prefer `tokio::Notify` if we don't actually need `'static`.

Reply:
Enabling `'static` futures would definitely be desirable from a user perspective.

Reply:
Thanks for the feedback! Can you provide a list of the specific futures you'd find that useful for? For example, would you want it for the read/write futures as well?

Reply:
Actually, I might have to take back what I said above. For new outgoing connections we need `'static` futures (which quinn already provides with the `Connecting` future). Futures for new streams on the other hand are driven in the same struct that owns the `Connection`, so I don't think the futures need to be 'static. However, I did not look into it in detail yet; as written in https://github.com/hyperium/h3/pull/80 it could still get complicated with the lifetimes. 
I'm sorry that I can not give more info at the current point; we are currently undergoing some internal refactoring of how we handle transports and multiplexing, and things are still in discussion. 
Either way, until #1219 is merged we'll probably stick with using `quinn_proto` anyway. So I don't think our use-case has to be considered right now, and thus using `Tokio's` Notify is sufficient. I am happy to do a PR to re-add `NotifyOwned` if it turns out that it's needed from our side. But of course I can't speak for the h3 folks and other users.

Reply:
Thanks for elaborating! I'll probably leave this open for now in to gather further feedback. I think we can release 0.9 without it and roll back to named futures without breaking, even.

Reply:
Went with tokio's `Notify` in the end because GATs are stabilizing and we haven't gotten any feedback clearly requesting `'static` regardles.